### PR TITLE
Off switch

### PR DESCRIPTION
**Pull request by lukechampine**
This is a replacement of #918.

I waffled on adding a separate command to siac, because I wasn't sure what to name it. If you come up with a good name, I'll add it.

### CODE REVIEW

**Comment by VoidingWarranties** [comment]
What about `siac host pause` and `siac host resume`?

**Comment by lukechampine** [comment]
I like those better than `siac host on`/`siac host off`. Let's see what david thinks

**Comment by DavidVorick** [comment]
it's better than on/off, but pause/resume is still poorly named - the host when 'paused' will still service uploads and downloads, and will still respond to other types of requests as well

File: doc/API.md [hunk 1]
```diff
@@ -336,10 +336,19 @@ struct {
 	unlockhash   types.UnlockHash  (string)
 	windowsize   types.BlockHeight (uint64)
 
+	acceptingcontracts bool
 	numcontracts       uint64
 	revenue            types.Currency (string)
 	storageremaining   int64
 	anticipatedrevenue types.Currency (string)
```
Comment (DavidVorick) [inline doc/API.md]: clusters should be alphabetically sorted, previous PR screwed this up, can you fix it?

File: modules/host.go [hunk 2]
```diff
@@ -52,14 +52,15 @@ type (
...
-		UnlockHash   types.UnlockHash  `json:"unlockhash"`
+		AcceptingContracts bool              `json:"acceptingcontracts"`
+		NetAddress         NetAddress        `json:"netaddress"`
+		TotalStorage       int64             `json:"totalstorage"`
+		MinDuration        types.BlockHeight `json:"minduration"`
+		MaxDuration        types.BlockHeight `json:"maxduration"`
+		WindowSize         types.BlockHeight `json:"windowsize"`
+		Price              types.Currency    `json:"price"`
+		Collateral         types.Currency    `json:"collateral"`
+		UnlockHash         types.UnlockHash  `json:"unlockhash"`
```
Comment (DavidVorick) [inline modules/host.go]: These should be sorted

File: modules/host/announce.go [hunk 3]
```diff
@@ -78,5 +83,8 @@ func (h *Host) AnnounceAddress(addr modules.NetAddress) error {
 	if h.closed {
 		return errHostClosed
 	}
+
+	h.mu.Lock()
+	defer h.mu.Unlock()
```
Comment (DavidVorick) [inline modules/host/announce.go]: probably okay to throw this to the top, other functions in the host take the form

```
lock
defer unlock
resourcelock
defer resource unlock
if closed {
}

rest of function
```

**Comment by DavidVorick** [comment]
I think that the current siac implementation is okay.

File: modules/host.go (hunk 2 above)
Comment (lukechampine) [inline modules/host.go]: why

Comment (lukechampine) [inline modules/host.go]: lol why

**Comment by VoidingWarranties** [comment]
If using the off switch won't be a common need then having a siac command for it is unnecessary. What are the common reasons a host would want to use the off switch?

**Comment by DavidVorick** [comment]
Pretty much the only reason that someone would turn off the host is if they are planning on repurposing the storage at some point in the future. They'd want to finish out their current contracts and then get the space back.

I don't think it's a common use case. Siac should be able to disable the host, but I don't think a separate command is necessary.

**Comment by lukechampine** [comment]
Agreed. I think `siac host set acceptingcontracts false` is sufficiently intuitive. Of course, it will be more user-friendly in Sia-UI.

**Comment by DavidVorick** [comment]
`siac host set` sounds fine to me.

**Comment by DavidVorick** [comment]
though we might want to think about solidifying the `siac` commands along with the api, to preserve compatibility with bash scripts and stuff

File: modules/host.go (hunk 2 above)
Comment (DavidVorick) [inline modules/host.go]: It makes the fields easier to find. I think every grouping should be sorted internally, and if you feel like there's a reason to break the sorting, then that's reason enough to break apart the grouping.

**Pull request merged**